fe/fe-core/src/main/java/org/apache/doris/planner/external/ExternalFileScanNode.java

```diff
@@ -202,7 +208,16 @@ public void init(Analyzer analyzer) throws UserException {
                 throw new UserException("Unknown type: " + type);
         }
 
-        backendPolicy.init();
+        if (isLocalTVF) {
+            // for local file, create a LocalTVFBackendPolicy which will always return the
+            // backend that user specify by "backend_id" = "xxx"
+            LocalTVFBackendPolicy tvfBackendPolicy = new LocalTVFBackendPolicy();
+            tvfBackendPolicy.init(backendId);
+            backendPolicy = tvfBackendPolicy;
+        } else {
+            backendPolicy = new BackendPolicy();
+            backendPolicy.init();
```

Comment:
This is not a good design, but this is not your problem.
we are working on refactoring the `split` interface here #17390,
maybe we can hold for a while.